docs/deprecation.md

```diff
@@ -39,5 +39,60 @@ In order to deprecate a feature gate the following conditions must be met:
+objects (e.g. Nodes). This metadata is exposed to the end-user which might use it and depend on it.
+
+Usually, metadata needs to be deprecated if it has a non-intuitive / irrelevant name or reflects data that is no
+longer relevant / useful.
+
+#### Deprecation
+In order to deprecate metadata the following conditions must be met:
+* A mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform about the deprecation.
+* In case the metadata is being renamed / changed, the new metadata (e.g. label) should be added alongside the deprecated
+one.
+
+For example, if we want to deprecate a label named `kubevirt.io/badNonIntuitiveName` to
+`kubevirt.io/greatNewName`, both labels should appear.
+
+#### Removal
+Deprecated metadata will not be removed for at least 3 release cycles, which means roughly 1 year.
+
+Afterwards, a mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform
+about the removal. If there are no objections, the metadata can be removed. Otherwise, a new removal date
+can be discussed according to the context.
+
+### API
+Although the following only mentions API objects, it's also relevant for their APIs (e.g. object fields),
+Kubevirt CR configuration, etc. Obviously, for fields / configs deprecation the step about adding a
+deprecated annotations can be skipped. However, a warning should always be triggered for any usage of all
+of the above mentioned.
+
+#### Deprecation
+In order to deprecate an API object the following actions need to be taken:
+* The deprecated object needs to contain a warning about the object being deprecated. The warning should also
+  state whenever the object would be removed (see below).
+* A `"kubevirt.io/deprecated":"true"` annotation should be added to all of the object instances.
+* A mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform about the deprecation.
+
+#### Removal
+A deprecated object will not be removed for at least 5 release cycles, which means roughly 1.5 years.
+
+Afterwards, a mail should be sent to kubevirt-dev mailing list (kubevirt-dev@googlegroups.com) to inform
+about the removal. If there are no objections, the object can be removed. Otherwise, a new removal date
+can be discussed according to the context.
+
+### Feature Behavior
+This refers to any changes in features behavior that might surprise the user or break backwards compatibility.
+
+As an example, let's say that a certain feature can be enabled via Kubevirt CR that affects only worker nodes,
+and that feature is changed to also affect control-plane nodes. This is considered a feature behavior change.
+Internal details that are not exposed to the end-user as guarantees that are part of our API aren't considered
+as feature behavior changes.
+
+The process is identical to the process of deprecating API (see above), except the number of releases before
+the final behavior change should be 3 releases (roughly 1 year).
```

Comment:
Should we take into account the state of the Feature/API? Currently, this doc does not differentiate between alpha, beta, v1. Perhaps worth elaborating on that explicitly?

Reply:
Thank you! A great point indeed.
Let's move the discussion to https://github.com/kubevirt/kubevirt/pull/9012#discussion_r1229514274